tel2puml/find_unique_graphs/otel_ingestion/otel_data_holder.py

```diff
@@ -67,10 +92,40 @@ def __init__(self, config: SQLDataHolderConfig) -> None:
         self.node_models_to_save: list[NodeModel] = []
         self.node_relationships_to_save: list[dict[str, str]] = []
         self.batch_size: int = config["batch_size"]
-        self.engine: Engine = create_engine(config["db_uri"], echo=False)
+        self.config = config
+
+    def __enter__(self) -> Self:
+        """Configure the database engine, session and create database tables.
+
+        :return: The object itself
+        :rtype: :class:`Self`
+        """
+
+        self.engine: Engine = create_engine(self.config["db_uri"], echo=False)
```

Comment:
Don't think you need this here really as it can all be done in init and it is preferred to set attributes there. Don't have to use it if we don't need to